### PR TITLE
Add wall mounted lockers

### LINKED ISSUE #10647: Add wallmount lockers
## Description

Lockers that are mounted on a wall. Good for conserving space on shuttles and will be nice for forks to have. Simply make a wall mount with entitystorage that has an offset (see the morgue for how to do this).

### PR DESCRIPTION
**Pull request by martin69420**
Closes #10647
Adds wall mounted lockers

**Screenshots**
wall-locker

**Changelog**

:cl:
- add: Added wall lockers

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+- type: entity
+  id: WallLockerBase
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml:2]:
```suggestion
  id: BaseWallLocker
```

File: Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  placement:
+    mode: SnapgridCenter
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Transform
+    noRot: true
+  - type: InteractionOutline
+  - type: WallMount
+    arc: 360
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml:12]: Should they work on the other side?

File: Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+- type: entity
+  id: WallLockerBase
+  placement:
+    mode: SnapgridCenter
+  abstract: true
+  components:
+  - type: Clickable
+  - type: Transform
+    noRot: true
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml:9]: I don't think this should be noRot?

File: Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml (hunk 1 above)
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Storage/Closets/WallLockers/base_structurewalllockers.yml:2]: Just put this in the same file as wall_lockers.yml and move it up a directory.

File: Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "https://github.com/space-wizards/space-station-14/pull/10673",
```
Comment (metalgearsloth) [inline Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json:4]: Did you make this?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json (hunk 4 above)
Comment (martin69420) [reply Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json:4]: Yes I did, but it's based off of one of the locker door textures

File: Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      "name": "door"
+    },
+    {
+      "name": "locked"
+    },
+    {
+      "name": "unlocked"
+    },
+    {
+        "name": "welded"
```
Comment (metalgearsloth) [inline Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json:20]:
```suggestion
      "name": "welded"
```

File: Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json (hunk 4 above)
Comment (metalgearsloth) [reply Resources/Textures/Structures/Storage/wall_locker.rsi/meta.json:4]: Add in a "modified from XYZ" then to attribute the source.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**